experimental/casemapping/src/casemapping.rs

```diff
@@ -114,19 +152,73 @@ impl CaseMapping {
     /// This function only implements simple and common mappings. Full mappings,
     /// which can map one `char` to a string, are not included.
     /// For full mappings, use [`CaseMapping::to_full_uppercase`].
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use icu_casemapping::CaseMapping;
+    ///
+    /// let cm = CaseMapping::new();
+    ///
+    /// assert_eq!(cm.to_uppercase('c'), 'C');
+    /// assert_eq!(cm.to_uppercase('C'), 'C');
+    /// assert_eq!(cm.to_uppercase('ć'), 'Ć');
+    /// assert_eq!(cm.to_uppercase('γ'), 'Γ');
+    ///
+    /// assert_eq!(cm.to_uppercase('ǳ'), 'Ǳ');
+    /// ```
     pub fn to_uppercase(&self, c: char) -> char {
         self.data.get().simple_upper(c)
     }
 
     /// Returns the titlecase mapping of the given `char`.
     /// This function only implements simple and common mappings. Full mappings,
     /// which can map one `char` to a string, are not included.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use icu_casemapping::CaseMapping;
+    ///
+    /// let cm = CaseMapping::new();
+    ///
+    /// assert_eq!(cm.to_titlecase('ǳ'), 'ǲ');
+    ///
+    /// assert_eq!(cm.to_titlecase('c'), 'C');
+    /// assert_eq!(cm.to_titlecase('C'), 'C');
+    /// assert_eq!(cm.to_titlecase('ć'), 'Ć');
+    /// assert_eq!(cm.to_titlecase('γ'), 'Γ');
+    /// ```
     pub fn to_titlecase(&self, c: char) -> char {
         self.data.get().simple_title(c)
     }
 
     /// Returns the simple case folding mapping of the given char.
     /// For full mappings, use [`CaseMapping::full_fold`].
+    ///
+    /// You can use the case folding to compare characters case insensitively.
```

Comment:
I think the statement being made here is that two single-character strings are caseless matches (as defined by D44 of TUS) if and only if the result of `fold` is the same for the two characters; in other words:
```rs
(cm.fold(x) == cm.fold(y)) ==
(cm.full_fold_string(x.to_string()) == cm.full_fold_string(y.to_string()))
```

**However:** This is not true until Unicode Version 15.1, see [UTC-175-C19](https://www.unicode.org/L2/L2023/23076.htm#175-C19).

Counterexample before 15.1:
```rs
let x = 'ﬅ';
let y = 'ﬆ';
```

Reply:
I can tweak this to say "provided this case folds to a single codepoint" since there's already a caveat about this

Reply:
How does this look?

Reply:
(Replied in a more up-to-date diff.)